src/app/components/search/CustomSliderComponent.tsx

```diff
@@ -0,0 +1,145 @@
+import HorizontalRuleIcon from "@mui/icons-material/HorizontalRule";
+import { Box, Slider, Stack, Typography } from "@mui/joy";
+import React, { useEffect } from "react";
+
+import MinMaxInputComponent from "./MinMaxInputComponent";
+
+function valueText(value: number) {
+  return `${value}`;
+}
+
+interface CustomSliderComponentProps {
+  onSliderChange: (values: [number, number]) => void;
+  defaultRangeValue: [number, number];
+  sliderRange: number[];
+  marks?: { value: number; label: string }[];
+  showMinMaxInputs?: boolean;
+  label: string;
+}
+
+export default function CustomSliderComponent({
+  onSliderChange,
+  defaultRangeValue,
+  sliderRange,
+  marks,
+  label,
+  showMinMaxInputs = true,
+}: CustomSliderComponentProps) {
+  const [range, setRange] = React.useState<number[]>([
+    defaultRangeValue[0],
+    defaultRangeValue[1],
+  ]);
+  const [minValue, setMinValue] = React.useState(
+    defaultRangeValue[0].toString(),
+  );
+  const [maxValue, setMaxValue] = React.useState(
+    defaultRangeValue[1].toString(),
+  );
+
+  useEffect(() => {
+    setRange([sliderRange[0], sliderRange[1]]);
+    setMinValue(sliderRange[0].toString());
+    setMaxValue(sliderRange[1].toString());
+  }, [sliderRange]);
+
+  const handleChange = (event: Event, newValue: number | number[]) => {
+    setRange(newValue as number[]);
+  };
+
+  // Update slider value when min or max input field changes
+  const handleMinInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setMinValue(e.target.value);
+  };
+
+  const handleMaxInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setMaxValue(e.target.value);
+  };
+
+  const handleInputKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      const newMinValue = parseInt(minValue);
+      const newMaxValue = parseInt(maxValue);
+
+      // Check if values are within valid range and min < max
+      if (newMinValue <= newMaxValue) {
```

Comment:
Negate this and make a guard clause like:
```js
if (newMinValue > newMaxValue)
e.preventDefault()
```
 So that you don't have to nest unnecessarily